tests/gtfs/test_validation.py

```diff
@@ -580,3 +602,203 @@ def test_summarise_routes_on_pass(self, gtfs_fixture):
+            ValueError,
+            match=re.escape(
+                "'orientation expected 'v' or 'h'"
+                " (non case sensitive)."
+                " Found l of type <class 'str'>."
+            ),
+        ):
+            current_fixture._plot_summary(
+                current_fixture.daily_route_summary,
+                "route_count_mean",
+                orientation="l",
+            )
+
+    @pytest.mark.filterwarnings("ignore::UserWarning")
+    def test__plot_summary_on_pass(self, gtfs_fixture, tmp_path):
+        """Test plotting a summary when defences are passed."""
+        current_fixture = gtfs_fixture
+        current_fixture.summarise_routes()
+
+        # test returning a html string
+        test_html = gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary,
+            "route_count_mean",
+            return_html=True,
+        )
+        assert type(test_html) == str, "Failed to return HTML for the plot"
+
+        # test returning a plotly figure
+        test_image = gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary, "route_count_mean"
+        )
+        assert (
+            type(test_image) == PlotlyFigure
+        ), "Failed to return plotly.graph_objects.Figure type"
+
+        # test saving plots in html and png format
+        gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary,
+            "route_count_mean",
+            width=1200,
+            height=800,
+            save_html=True,
+            save_image=True,
+            ylabel="Mean",
+            xlabel="Day",
+            orientation="h",
+            plotly_kwargs={"legend": dict(bgcolor="lightgrey")},
+            out_dir=os.path.join(tmp_path, "save_test"),
+        )
+
+        # general save test
+        save_dir = os.listdir(os.path.join(tmp_path, "save_test"))
+        counts = {"html": 0, "png": 0}
+        for pth in save_dir:
+            if ".html" in pth:
+                counts["html"] += 1
+            elif ".png" in pth:
+                counts["png"] += 1
+
+        assert os.path.exists(
+            os.path.join(tmp_path, "save_test")
+        ), "'save_test' dir could not be created'"
+        assert counts["html"] == 1, "Failed to save plot as HTML"
+        assert counts["png"] == 1, "Failed to save plot as png"
+
+        # save test for an image with invalid file extension
+        valid_img_formats = ["png", "pdf", "jpg", "jpeg", "webp", "svg"]
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Please specify a valid image format. Valid formats "
+                f"include {valid_img_formats}"
+            ),
+        ):
+            gtfs_fixture._plot_summary(
+                gtfs_fixture.daily_route_summary,
+                "route_count_mean",
+                save_image=True,
+                out_dir=os.path.join(tmp_path, "outputs"),
+                img_type="test",
+            )
```

Comment:
This looks like it belongs with `test__plot_summary_defences` not `test__plot_summary_on_pass`